Add basic Cypress end-to-end testing

abelsiqueira opened the pull request:
# Pull request details

## List of related issues or pull requests

Refs:
- #610 

## Describe the changes made in this pull request

Includes the minimum for Cypress testing to work and an example filling the form with minimum information and checking the CITATION.cff.

## Instructions to review the pull request

Run cypress locally:

```shell
cd $(mktemp -d --tmpdir cffinit-pr.XXXXXX)
git clone https://github.com/citation-file-format/cff-initializer-javascript .
git checkout <this branch>
npm clean-install
npm run dev
npm run cypress:open
```

abelsiqueira committed:
Add basic Cypress end-to-end testing

fdiblen commented on cypress/fixtures/example.json:
```diff
@@ -0,0 +1,5 @@
+{
```
Can you delete this file as it is not used.

fdiblen commented on cypress/support/commands.ts:
```diff
@@ -0,0 +1,37 @@
+/// <reference types="cypress" />
```
Can you delete this file as it is not used.

fdiblen commented on README.dev.md:
```diff
@@ -65,6 +67,20 @@ You can also use the Majestic web interface to run the unit tests in your browse
+
+Select "E2E Testing", which should be configured already and select the desired browser and click the start button.
+
+A browser will open with the list of tests, which can be explored.
```
Can you please add the next steps to be able to run the tests? If you have time, it would be nice to have screenshots.

fdiblen commented on cypress/support/e2e.ts:
```diff
@@ -0,0 +1,20 @@
+// ***********************************************************
```
can you remove commented lines?

fdiblen reviewed (changes requested):
It works nicely. Thank you @abelsiqueira. I suggested a few minor changes. Can you also pull the changes from the main branch? This will fix the issue about unit tests.

abelsiqueira committed:
Implements @fdiblen's suggestions from the PR review

Add more information to README.dev.md
Remove cypress/support and cypress/fixtures

abelsiqueira commented:
Thanks for the review, Faruk. I've removed the two files you mentioned, and an additional one that only existed because of those two. I made the configuration change to ignore the missing file.
I've added more information to the README. I wanted to avoid adding images, so I just linked to one on the official Cypress website.

fdiblen reviewed (approved):
Looks great. Thanks @abelsiqueira  :brazil: :brazil: :brazil:
